pika/adapters/base_connection.py

```diff
@@ -315,63 +304,57 @@ def _flush_outbound(self):
         """
-        # TODO doesn't seem right: docstring defines error_value as int|object,
-        # but _get_error_code expects a falsie or an exception-like object
-        error_code = self._get_error_code(error_value)
-
-        if not error_code:
-            LOGGER.critical("Tried to handle an error where no error existed")
-            return
-
-        # Ok errors, just continue what we were doing before
-        if error_code in self.ERRORS_TO_IGNORE:
-            LOGGER.debug("Ignoring %s", error_code)
-            return
+        # Assertion failure here would indicate internal logic error
+        if not exception.errno:
+            raise ValueError("No socket error indicated: %r" % (exception,))
+        if exception.errno in self.ERRORS_TO_IGNORE:
+            raise ValueError("Soft error not handled by I/O logic: %r"
+                             % (exception,))
 
         # Socket is no longer connected, abort
-        elif error_code in self.ERRORS_TO_ABORT:
-            LOGGER.error("Fatal Socket Error: %r", error_value)
+        if exception.errno in self.ERRORS_TO_ABORT:
+            LOGGER.error("Fatal Socket Error: %r", exception)
 
-        elif self.params.ssl and isinstance(error_value, ssl.SSLError):
+        elif self.params.ssl and isinstance(exception, ssl.SSLError):
 
-            if error_value.args[0] == ssl.SSL_ERROR_WANT_READ:
+            if exception.errno == ssl.SSL_ERROR_WANT_READ:
                 # TODO doesn't seem right: this logic updates event state, but
                 # the logic at the bottom unconditionaly disconnects anyway.
+                # Clearly, we're not prepared to handle re-handshaking. It
+                # should have been handled gracefully without ending up in this
+                # error handler; ditto for SSL_ERROR_WANT_WRITE below.
                 self.event_state = self.READ
-            elif error_value.args[0] == ssl.SSL_ERROR_WANT_WRITE:
+            elif exception.errno == ssl.SSL_ERROR_WANT_WRITE:
                 self.event_state = self.WRITE
             else:
-                LOGGER.error("SSL Socket error: %r", error_value)
+                LOGGER.error("SSL Socket error: %r", exception)
 
         else:
             # Haven't run into this one yet, log it.
-            LOGGER.error("Socket Error: %s", error_code)
+            LOGGER.error("Unexpected socket Error: %r", exception)
 
         # Disconnect from our IOLoop and let Connection know what's up
         self._on_terminate(connection.InternalCloseReasons.SOCKET_ERROR,
-                           repr(error_value))
+                           repr(exception))
 
     def _handle_timeout(self):
         """Handle a socket timeout in read or write.
         We don't do anything in the non-blocking handlers because we
         only have the socket in a blocking state during connect."""
         LOGGER.warning("Unexpected socket timeout")
 
-    def _handle_events(self, fd, events, error=None, write_only=False):
```

Comment:
`write_only` was most likely a vestige of the original `BlockingConnection` implementation prior to my rewrite of `BlockingConnection`.